Sources/PackagePlugin/PluginMessages.swift

```diff
@@ -0,0 +1,316 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021-2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
```

Comment:
This pulls together all the host-to-plugin messages and plugin-to-host messages, and the structures they referenced.  This source file is shared between the PackagePlugin module and SwiftPM code so that there is only one source of truth about the Codable protocol and we get compiler detection of mismatches.